solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -227,12 +405,82 @@ pub struct Deliver<'info> {
+    chain: Box<Account<'info, chain::ChainData>>,
+
+    system_program: Program<'info, System>,
+}
+
+#[derive(Accounts)]
+#[instruction(port_id: PortId, channel_id: ChannelId, base_denom: String)]
+pub struct MockDeliver<'info> {
+    #[account(mut)]
+    sender: Signer<'info>,
+
+    /// The account holding private IBC storage.
+    #[account(init_if_needed, payer = sender, seeds = [SOLANA_IBC_STORAGE_SEED],bump, space = 10000)]
+    storage: Account<'info, storage::PrivateStorage>,
+
+    /// The account holding provable IBC storage, i.e. the trie.
+    ///
+    /// CHECK: Account’s owner is checked by [`storage::get_provable_from`]
+    /// function.
+    #[account(init_if_needed, payer = sender, seeds = [TRIE_SEED], bump, space = 1000)]
+    trie: UncheckedAccount<'info>,
 
+    /// The account holding packets.
+    #[account(init_if_needed, payer = sender, seeds = [PACKET_SEED], bump, space = 1000)]
+    packets: Box<Account<'info, IBCPackets>>,
+
+    /// The below accounts are being created for testing purposes only.
+    /// In real, we would run conditionally create an escrow account when the channel is created.
+    /// And we could have another method that can create a mint given the denom.
+    #[account(init_if_needed, payer = sender, seeds = [MINT_ESCROW_SEED], bump, space = 100)]
+    /// CHECK:
+    mint_authority: UncheckedAccount<'info>,
+    #[account(init_if_needed, payer = sender, seeds = [base_denom.as_bytes().as_ref()], bump, mint::decimals = 6, mint::authority = mint_authority)]
+    token_mint: Account<'info, Mint>,
+    #[account(init_if_needed, payer = sender, seeds = [port_id.as_bytes().as_ref(), channel_id.as_bytes().as_ref()], bump, token::mint = token_mint, token::authority = sender)]
+    escrow_account: Box<Account<'info, TokenAccount>>,
+    #[account(init_if_needed, payer = sender, associated_token::mint = token_mint, associated_token::authority = sender)]
+    sender_token_account: Box<Account<'info, TokenAccount>>,
+
+    associated_token_program: Program<'info, AssociatedToken>,
+    token_program: Program<'info, Token>,
     system_program: Program<'info, System>,
 }
 
-impl Router for storage::IbcStorage<'_, '_> {
+/// Error returned when handling a request.
+#[derive(Clone, strum::AsRefStr, strum::EnumDiscriminants)]
+#[strum_discriminants(repr(u32))]
+pub enum Error<'a> {
+    RouterError(&'a ibc::core::RouterError),
+}
+
+impl Error<'_> {
+    pub fn name(&self) -> String { self.as_ref().into() }
+}
+
+impl core::fmt::Display for Error<'_> {
+    fn fmt(&self, fmtr: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            Self::RouterError(err) => write!(fmtr, "{err}"),
+        }
+    }
+}
+
+impl From<Error<'_>> for u32 {
+    fn from(err: Error<'_>) -> u32 {
+        let code = ErrorDiscriminants::from(err) as u32;
+        anchor_lang::error::ERROR_CODE_OFFSET + code
+    }
+}
+
+#[event]
+pub struct EmitIBCEvent {
+    pub ibc_event: Vec<u8>,
+}
+
```

Comment:
```suggestion
```

Those now live in error.rs and events.rs.  This looks like left over from previous version of the code.

Reply:
understood